elyra/pipeline/component.py

```diff
@@ -171,12 +172,12 @@ class Component(object):
     runtime: str
     op: str
     properties: List[ComponentParameter]
-    extensions: str
+    extensions: Optional[List[str]]
```

Comment:
Why is this becoming a list? Are we assuming that, if multiple extensions, the "running" aspects would still be same?

Reply:
IIRC, this was made a list since it (ultimately) corresponds to a list-valued attribute in what becomes the palette.json under `app_data.extensions:`.  I think the only assumption is that it enables support for a list of extensions, rather than one, and matches the frontend expectations (of a list).

> if multiple extensions, the "running" aspects would still be same?

 Since `Component` is really a client-side (pre-execution) thing used to support drag and drop, I actually don't think this comes into play.  `Component`'s runtime counter-part is the `Operation` (`GenericOperation`) - which doesn't reflect anything about the extension since it would reference the actual file itself.